packages/runtime/src/widgets/Conditional.tsx

```diff
@@ -0,0 +1,96 @@
+import { Expression, Widget, useExecuteCode } from "framework";
+import { WidgetRegistry } from "../registry";
+import { EnsembleRuntime } from "../runtime";
+import { handleCurlyBraces } from "../util/utils";
+import { head, isEmpty, last } from "lodash-es";
+
+type CondtionalElement = Record<Capitalize<string>, Widget> &
+  (
+    | { if: Expression<string>; elseif?: never; else?: never }
+    | { elseif: Expression<string>; if?: never; else?: never }
+    | { else: null; if?: never; elseif?: never }
+  );
+
+export type ConditionalProps = {
+  conditions: CondtionalElement[];
+};
+
+export const Conditional: React.FC<ConditionalProps> = (props) => {
+  const [isValid, errorMessage] = hasProperStructure(props.conditions);
+  if (!isValid) throw Error(errorMessage);
+
+  let element = head(
+    props.conditions.filter((condition) => {
```

Comment:
You can use `some` to avoid iterating over entire list.

Reply:
Yeah you can use the `evaluate` util instead of the hook to get the value immediately